src/Razor/src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs

```diff
@@ -94,68 +106,90 @@ internal void TriggerSmartIndent(ITextView textView)
         }
 
         // Internal for testing
-        internal void TextBuffer_OnChanged(object sender, TextContentChangedEventArgs args)
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        internal async void TextBuffer_OnChanged(object sender, TextContentChangedEventArgs args)
+#pragma warning restore VSTHRD100 // Avoid async void methods
         {
-            _dispatcher.AssertForegroundThread();
-
-            if (!args.TextChangeOccurred(out var changeInformation))
+            try
             {
-                return;
-            }
+                await _foregroundDispatcher.RunOnForegroundAsync(() =>
```

Comment:
Ah IIRC this can run on the UI thread like it did before since I don't believe it interacts with our ProjectSnapshotManager